zebra-script/src/lib.rs

```diff
@@ -2,10 +2,155 @@
 #![doc(html_logo_url = "https://www.zfnd.org/images/zebra-icon.png")]
 #![doc(html_root_url = "https://doc.zebra.zfnd.org/zebra_script")]
 
+use displaydoc::Display;
+use thiserror::Error;
+use zcashconsensus::{
+    zcashconsensus_error_t, zcashconsensus_error_t_zcashconsensus_ERR_OK,
+    zcashconsensus_error_t_zcashconsensus_ERR_TX_DESERIALIZE,
+    zcashconsensus_error_t_zcashconsensus_ERR_TX_INDEX,
+    zcashconsensus_error_t_zcashconsensus_ERR_TX_SIZE_MISMATCH,
+};
+use zebra_chain::{
+    block::Block,
+    transaction::{TransparentInput, TransparentOutput},
+};
+
+#[derive(Debug, Display, Error)]
+#[non_exhaustive]
+pub enum Error {
+    /// Verification failed but tx was valid
+    #[non_exhaustive]
+    ErrOk,
```

Comment:
I'm a little confused by `ErrOk`.  What does this error mean.  Do we know already, or is this translating error codes from the C++ implementation?

Reply:
its a straight translation, I think the doc comment is accurate to what it means but yea, what a fking name...

Reply:
Do we know what errors the function can actually generate? I think we should not propagate `ErrOk` up to the users of the API.

Reply:
afaik this is the exact set of errors the function generates. We can make the specific variant an implementation detail that we wrap in an opaque error struct and fix it later if we need to match for specific error kinds.

Reply:
AFAICT this should just be treated as "script failed to verify".

Reply:
`ErrOk` is a horrible name and you have my endorsement to kill it with fire.

Reply:
I've renamed it on our side at least,